source/state_representation/test/tests/space/cartesian/test_cartesian_pose.cpp

```diff
@@ -0,0 +1,277 @@
+
+TEST(CartesianPoseTest, RandomPoseInitialization) {
+  CartesianPose random = CartesianPose::Random("test");
+  // only position should be random
+  EXPECT_GT(random.get_position().norm(), 0);
+  EXPECT_GT(abs(random.get_orientation().w()), 0);
+  EXPECT_GT(abs(random.get_orientation().x()), 0);
+  EXPECT_GT(abs(random.get_orientation().y()), 0);
+  EXPECT_GT(abs(random.get_orientation().z()), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(random).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(random).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(random).get_wrench().norm(), 0);
+}
+
+TEST(CartesianPoseTest, CopyPose) {
+  CartesianPose pose1 = CartesianPose::Random("test");
+  CartesianPose pose2(pose1);
+  EXPECT_EQ(pose1.get_name(), pose2.get_name());
+  EXPECT_EQ(pose1.get_reference_frame(), pose2.get_reference_frame());
+  EXPECT_TRUE(pose1.data().isApprox(pose2.data()));
+  EXPECT_EQ(static_cast<CartesianState&>(pose2).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose2).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose2).get_wrench().norm(), 0);
+  CartesianPose pose3 = pose1;
+  EXPECT_EQ(pose1.get_name(), pose3.get_name());
+  EXPECT_EQ(pose1.get_reference_frame(), pose3.get_reference_frame());
+  EXPECT_TRUE(pose1.data().isApprox(pose3.data()));
+  EXPECT_EQ(static_cast<CartesianState&>(pose3).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose3).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose3).get_wrench().norm(), 0);
+  // try to change non pose variables prior to the copy, those should be discarded
+  static_cast<CartesianState&>(pose1).set_twist(Eigen::VectorXd::Random(6));
+  static_cast<CartesianState&>(pose1).set_accelerations(Eigen::VectorXd::Random(6));
+  static_cast<CartesianState&>(pose1).set_wrench(Eigen::VectorXd::Random(6));
+  CartesianPose pose4 = pose1;
+  EXPECT_TRUE(pose1.data().isApprox(pose4.data()));
+  EXPECT_EQ(static_cast<CartesianState&>(pose4).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose4).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose4).get_wrench().norm(), 0);
+  // copy a state, only the pose variables should be non 0
+  CartesianPose pose5 = CartesianState::Random("test");
+  EXPECT_EQ(static_cast<CartesianState&>(pose5).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose5).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose5).get_wrench().norm(), 0);
+
+  CartesianPose pose6;
+  EXPECT_TRUE(pose6.is_empty());
+  CartesianPose pose7 = pose6;
+  EXPECT_TRUE(pose7.is_empty());
+}
+
+TEST(CartesianPoseTest, SetData) {
+  CartesianPose cp1 = CartesianPose::Identity("test");
+  CartesianPose cp2 = CartesianPose::Random("test");
+  cp1.set_data(cp2.data());
+  EXPECT_TRUE(cp2.data().isApprox(cp1.data()));
+
+  auto pose_vec = cp2.to_std_vector();
```

Comment:
I think `cp1` should probably be reset before this line actually, and / or `cp2` re-randomized, just to properly test the other signature.